src/Control/Monad/Free.hs

```diff
@@ -71,6 +71,14 @@ import Prelude hiding (foldr)
 import GHC.Generics
 #endif
 
+-- $setup
+-- >>> import Control.Applicative (Const (..))
+-- >>> import Data.Functor.Identity (Identity (..))
+-- >>> import Data.Monoid (First (..))
+-- >>> import Data.Tagged (Tagged (..))
+-- >>> let preview l x = getFirst (getConst (l (Const . First . Just) x))
```

Comment:
Since `preview` and `review` are defined separately in the doctests for two different modules, I wonder if it's worthwhile to define an internal module for these functions, similar to [`Numeric.AD.Internal.Doctest`](https://github.com/ekmett/ad/blob/6edd6c6aa1df3a7b5f3d56b458ae09a228c297c3/src/Numeric/AD/Internal/Doctest.hs) in `ad`.

Reply:
It would need to be public module. I don't like that idea.

Reply:
Why would it need to be a public module? As long as the module exports `preview` and `review`, I don't see what the issue would be. (`Numeric.AD.Internal.Doctest` is an internal module, as another data point.)

Reply:
I mean public as in `exposed-modules`.

Reply:
Again, `Numeric.AD.Internal.Doctest` isn't an `exposed-module`, it's an `other-module`. I fail to see why we couldn't do the same thing in `free`.

Reply:
`ad` its not that bad, as there are plenty of internal modules. `free` is nice and clean atm.

Reply:
I'm still struggling to comprehend this at a technical level. Can you explain to me like I'm 5 years old why this isn't feasible?

Reply:
It is exposed module now: https://github.com/ekmett/ad/pull/90/files#diff-c1d4c5c5ca8626220e702ed03bbc62044ae6f5212eaa243ad09f8101f5a5bb74R135

I think `RDouble` is valuable enough (e.g. for  `hyperloglog` etc. numeric packages) that we should pull it into own package. Similarly here, we could embrace that the `review` and `preview` are from `lens` and depend on it. The problem with that is that `lens` depends on `free` so it won't work nicely, and e.g. `microlens` doesn't have `profunctors` based isomorphisms :(

Reply:
And to why it have to be exposed. Because `cabal-docspec` cannot import internal modules.

Reply:
> It is exposed module now: https://github.com/ekmett/ad/pull/90/files#diff-c1d4c5c5ca8626220e702ed03bbc62044ae6f5212eaa243ad09f8101f5a5bb74R135

Oh, this is news to me! It looks like this quietly changed in ekmett/ad#90, and I somehow didn't notice.

In any case, I wasn't aware of this `other-modules` restriction. (I was under the impression that `cabal-docspec` just loaded the object code directly from `.o` files, in which case the concept of `exposed-modules` versus `other-modules` would be irrelevant, but it appears there's more to it than that.) In light of this, I agree that copy-pasting the definitions of `preview` and `review`, while regrettable, is better than exposing a new module.

Reply:
`cabal-docspec` doesn't do anything fancy&special. Compare to `cabal-env package && ghci` and trying to play with that package.